model/pod/pod.go

```diff
@@ -87,3 +121,44 @@ func (p *Pod) Remove() error {
 		},
 	)
 }
+
+// UpdateStatus updates the pod status.
+func (p *Pod) UpdateStatus(s Status) error {
+	byIDAndStatus := ByID(p.ID)
+	byIDAndStatus[StatusKey] = p.Status
+
+	setFields := bson.M{
+		StatusKey: s,
+	}
+	statusUpdated := time.Now()
+	switch s {
+	case InitializingStatus:
```

Comment:
why do we need to save each of these times?

Reply:
Part of the design includes timing information (e.g. how long it takes to go from initializing to running). It's also useful for traceability purposes since the state is a transient field, whereas the timing information will be more stable (e.g. did a pod get terminated immediately after initializing, or did it make it to the running state first).

Reply:
I would have thought to track timing through logging to the event_log

Reply:
I guess we could do that instead, but figuring out exactly how to extract the thing you want from the event log is usually more painful.

Reply:
I don't have strong feelings about it. If you think we're better off keeping track of timing internally I'm all for it.

Reply:
I don't really feel strongly either way, so I'm fine with just leaving it as a TODO to add event logging and that's it.